packages/portfolio/src/portfolio-routes.tsx

```diff
@@ -1,17 +1,40 @@
-import { PortfolioRoutesLoaded } from './portfolio-routes-loaded.tsx'
-import { PortfolioUiAccountGuard } from './ui/portfolio-ui-account-guard.tsx'
-import { PortfolioUiNetworkGuard } from './ui/portfolio-ui-network-guard.tsx'
+import type { Account } from '@workspace/db/entity/account'
+import type { Network } from '@workspace/db/entity/network'
+import { useDbAccountLive } from '@workspace/db-react/use-db-account-live'
+import { useDbNetworkLive } from '@workspace/db-react/use-db-network-live'
+import { useDbSetting } from '@workspace/db-react/use-db-setting'
+import { useMemo } from 'react'
+import { useRoutes } from 'react-router'
+import { PortfolioFeatureIndex } from './portfolio-feature-index.tsx'
+import { PortfolioFeatureTx } from './portfolio-feature-tx.tsx'
+
+export interface AccountNetwork {
+  account: Account
+  network: Network
+}
 
 export default function PortfolioRoutes() {
-  return (
-    <PortfolioUiNetworkGuard
-      render={({ network }) => (
-        <PortfolioUiAccountGuard
-          render={({ account }) => {
-            return <PortfolioRoutesLoaded account={account} network={network} />
-          }}
-        />
-      )}
-    />
-  )
+  const networkLive = useDbNetworkLive()
+  const [activeId] = useDbSetting('activeNetworkId')
+  const network = useMemo(() => networkLive.find((item) => item.id === activeId), [activeId, networkLive])
+  if (!network) {
+    throw new Error('No active network set.')
+  }
+
+  const [accountId] = useDbSetting('activeAccountId')
+  const [walletId] = useDbSetting('activeWalletId')
+  if (!walletId) {
+    throw new Error('No active wallet set.')
+  }
+  const accountLive = useDbAccountLive({ walletId: walletId })
+  const account = useMemo(() => accountLive.find((item) => item.id === accountId), [accountId, accountLive])
+  if (!account) {
+    throw new Error('No active account set.')
+  }
+
+  return useRoutes([
+    { element: <PortfolioFeatureIndex account={account} network={network} />, path: '*' },
```

Comment:
The `network` and `account` here could be fetched from the hook mentioned in the comment on `tools-routes.tsx`

Reply:
Updated.